R/check_authorized_users.R

```diff
@@ -0,0 +1,85 @@
+#' Check if a user is authorized to modify model
+#' directories. This function verifies whether a GitHub
+#' user is authorized to modify specific model directories
+#' in a Hub by checking the designated users in model
+#' metadata.
+#'
+#' @param changed_dirs Character vector. Directory names
+#' that have been changed.
+#' @param gh_actor Character. GitHub username of the person
+#' making changes.
+#' @param base_hub_path Character. Path to the base hub
+#' directory.
+#'
+#' @return Exits invisibly, with error if the user is
+#' unauthorized and prints success message if authorized.
+#'
+#' @export
+check_authorized_users <- function(
+  changed_dirs,
+  gh_actor,
+  base_hub_path
+) {
+  checkmate::assert_character(changed_dirs, min.len = 1)
+  checkmate::assert_scalar(gh_actor)
```

Comment:
Should this not also be assert string if base_hub_path is?
```suggestion
  checkmate::assert_string(gh_actor)
```

Reply:
`gh_actor` should indeed be a scalar character, no?